tasks/dnsmasq.yml

```
@@ -61,12 +61,28 @@
         path: /etc/resolv.conf
       register: resolv_dot_conf
 
-    - name: Remove resolv.conf association with systemd-resolved
-      file:
-        src: /run/resolvconf/resolv.conf
-        path: /etc/resolv.conf
-        state: link
+    - block:
+        - name: Remove resolv.conf association with systemd-resolved
+          file:
+            path: /etc/resolv.conf
+            state: absent
+
+        - name: Create /etc/resolv.conf
+          file:
+            path: /etc/resolv.conf
+            state: touch
+            owner: root
+            group: root
+            mode: u=rw,g=r,o=r
+
+        - name: Add a nameserver entry poining to localhost for dnsmasq
+          lineinfile:
+            path: /etc/resolv.conf
+            regexp: "^nameserver 127.0.0.1"
+            line: "nameserver 127.0.0.1"
+            unsafe_writes: true  # to prevent failures in CI
       become: true
       when:
+        - resolv_dot_conf.stat.islnk is defined
```

This (`stat.islnk`) will always be defined, assuming that the file exists. If the idea here is to check if the file exists before examining `stat.islnk` and `stat.lnk_source`, then I think this would be more readable:

```suggestion
        - resolv_dot_conf.stat.exists
```

---

this is necessary to avoid the error of the condition "resolv_dot_conf.stat.islnk" if there is no link information in the variable resolv_dot_conf

---

> If this fix is indeed not necessary, please remove the two commits from the branch. Otherwise, what is necessary to fix RH compatibility?

I don't remember the details, maybe I fixed my mistake here. \
I'm not sure I know how to remove commit from history except to create a PR again.

---

> this is necessary to avoid the error of the condition "resolv_dot_conf.stat.islnk" if there is no link information in the variable resolv_dot_conf

Just out of curiosity, when does that happen? I've never seen `islnk` not be defined.

---

> > If this fix is indeed not necessary, please remove the two commits from the branch. Otherwise, what is necessary to fix RH compatibility?
> 
> I don't remember the details, maybe I fixed my mistake here. I'm not sure I know how to remove commit from history except to create a PR again.

You would need to rebase the branch and rewrite history. I can try to do that for you, but GitHub is being weird and not letting me push, even though it claims that maintainers can edit the branch.

---

thanks @nre-ableton